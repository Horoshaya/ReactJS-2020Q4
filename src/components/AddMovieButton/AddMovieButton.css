.button {
  padding: 10px;
  font-size: 18px;
  text-transform: uppercase;
  color: #f65261;
  background-color: #555555ab;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  transition-duration: 0.4s;
}

.button:hover {
  transform: scale(1.1);
}
