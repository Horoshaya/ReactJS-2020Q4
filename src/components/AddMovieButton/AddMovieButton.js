import React, { useState } from 'react';

import styles from './AddMovieButton.css';
import ModifyModal from '../ModifyModal/ModifyModal';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import { usePropsLogger } from '../../hooks/usePropsLogger';

const AddMovieButton = () => {
  const [isModalOpen, setModalOpen] = useState(false);

  usePropsLogger(null, 'AddMovieButton');

  const triggerModal = () => {
    setModalOpen(!isModalOpen);
  };

  return (
    <>
        <button className={styles.button} onClick={triggerModal}>
          + add movie
        </button>
      {isModalOpen ? (
        <ModifyModal triggerModal={triggerModal}></ModifyModal>
      ) : null}
    </>
  );
};

export default AddMovieButton;
