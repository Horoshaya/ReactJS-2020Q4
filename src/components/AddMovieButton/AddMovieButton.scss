.button {
  padding: 10px;
  font-size: 18px;
  text-transform: uppercase;
  color: #f65261;
  background-color: #555555ab;
  border: none;
  border-radius: 5px;
}
