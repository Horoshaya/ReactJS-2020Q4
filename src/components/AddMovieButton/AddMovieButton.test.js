import React from 'react';
import rerender from 'react-test-renderer';
import AddMovieButton from './AddMovieButton';
import { shallow } from 'enzyme';

describe('when AddButton', () => {
  test('then snapshot created', () => {
    const component = rerender.create(<AddMovieButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('openModal', () => {
    const component = shallow(<AddMovieButton />);
    component.triggerModal;
    const addButton = component.find('button');
    addButton.simulate('click');
    expect(addButton.text()).toEqual('+ add movie');
  });
});
