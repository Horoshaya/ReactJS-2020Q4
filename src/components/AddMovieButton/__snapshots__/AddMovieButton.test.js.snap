// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieButton should created snapshot 1`] = `
<button
  onClick={[Function]}
>
  + add movie
</button>
`;

exports[`when AddButton then snapshot created 1`] = `
<button
  onClick={[Function]}
>
  + add movie
</button>
`;
